greedy_max/__init__.py

```diff
@@ -43,20 +52,84 @@ def schedule_observation(self, site: Site, obs_idx: int, start: int, finish: int
         from start to finish.
         """
         self.schedule[site][start:finish] = obs_idx
+    
+    def schedule_observations(self, site: Site, science: List[Observation], start: int, end: int) -> NoReturn:
+        """
+        Add all the observations inside a unit in the time slots from start to finish.
+        """
+        for sci in science:
+            if start < end:
+                    sci_length = end - start+ 1  if start + sci.length - 1 > end else sci.length
+                    self.schedule_observation(site, sci.idx, start, start + sci_length)
+                    sci.observed+=sci_length 
+                    start+=sci_length
 
     def is_observation_scheduled(self, site: Site, obs_idx: int) -> bool:
         """
         Determine if an observation is partially or fully scheduled at the specified site.
         """
         return len(self.slots_for_observation(site, obs_idx)) != 0
 
+    def is_complete(self, site: Site) -> bool:
+        return len(self._empty_slots(site)) < 0
```

Comment:
Is `< 0` right or should it be `<= 0`?
(I have no idea... just seems like the `len` function would never return a negative number.)

Reply:
In fact, I'm thinking it should just be `== 0`, since `len` is never negative to my knowledge (and a google search), so as it stands ,I don't think this method ever returns anything but `False`.

Reply:
Even better:

```
return not self._empty_slots(site)
```

which returns `False` if empty, `True` if not empty. You'd have to modify the call signature to `is_not_empty` or similar.